test/verify/check-services

```diff
@@ -548,6 +549,88 @@ WantedBy=multi-user.target
 
         b.wait_not_present("#service-unit div.cond-fail")
 
+    def testRelationships(self):
+        m = self.machine
+        b = self.browser
+
+        m.write("/etc/systemd/system/test-a.service",
+"""
+[Unit]
+Description=A Service
+Before=test-b.service
+Conflicts=test-c.service
+
+[Service]
+ExecStart=/bin/sh -c "while true; do sleep 5; done"
+""")
+
+        m.write("/etc/systemd/system/test-b.service",
+"""
+[Unit]
+Description=B Service
+After=test-a.service
+
+[Service]
+ExecStart=/bin/sh -c "while true; do sleep 5; done"
+""")
+
+        m.write("/etc/systemd/system/test-c.service",
+"""
+[Unit]
+Description=C Service
+Conflicts=test-a.service
+PartOf=test-b.service
+
+[Service]
+ExecStart=/bin/sh -c "while true; do sleep 5; done"
+""")
+
+        # After writing files out tell systemd about them
+        m.execute("systemctl daemon-reload")
+
+        def svc_sel(service):
+            return 'tr[data-goto-unit="%s"]' % service
+
+        def rel_sel(reltype):
+            return "table.systemd-unit-relationship-table tr:has(td:contains('%s'))" % reltype
+
+        def wait_relationship(reltype, service):
+            b.wait_present(rel_sel(reltype))
+            b.wait_in_text(rel_sel(reltype), service)
+
+        def click_relationship(reltype, service):
+            b.click(rel_sel(reltype) + (" a:contains('%s')" % service))
+
+        # services page
+        self.login_and_go("/system/services")
+        b.wait_in_text("#services", "test-a.service")
+        b.click(svc_sel("test-a.service"))
+
+        # service a
+        b.wait_js_cond('window.location.hash === "#/test-a.service"')
+        wait_relationship("Before", "test-b.service")
+        wait_relationship("Conflicts", "test-c.service")
+        click_relationship("Before", "test-b.service")
+
+        # service b
+        b.wait_js_cond('window.location.hash === "#/test-b.service"')
```

Comment:
I like this "click-through" test. That plus asserting the hash here seem sufficient to me.